activerecord/lib/arel/visitors/to_sql.rb

```diff
@@ -784,14 +798,11 @@ def maybe_visit(thing, collector)
         end
 
         def inject_join(list, collector, join_str)
-          len = list.length - 1
-          list.each_with_index.inject(collector) { |c, (x, i)|
```

Comment:
>  ["~/rails/activerecord/lib/arel/visitors/to_sql.rb",
  788,
  :T_ARRAY]=>[3000, 0, 0, 0, 0, 0],

At least looks like this `each_with_index.inject` allocates three arrays.

Others are to use `unless i == 0` instead of `unless i == list.length - 1`.

> Is the each_with_index change related? I believe it may be slightly slower than manually tracking an index variable as before.

Do you mean that the befere (`unless i == list.length - 1`) is faster than the after (`unless i == 0`)?

Reply:
Thanks. I was on mobile and mistakenly thought you were adding an each_with_index where one did not previously exist.